projects/storefrontlib/src/cms-components/checkout/guards/checkout.guard.ts

```diff
@@ -7,47 +7,80 @@ import { RoutingConfigService } from '@spartacus/core';
 import { ExpressCheckoutService } from '../services/express-checkout.service';
 import { CheckoutConfigService } from '../services/checkout-config.service';
 import { CheckoutStepType } from '../model/checkout-step.model';
+import { CheckoutConfig } from '../config/checkout-config';
 
 @Injectable({
   providedIn: 'root',
 })
 export class CheckoutGuard implements CanActivate {
   private firstStep$: Observable<UrlTree>;
 
+  /**
+   * @deprecated since version 1.2
+   *  Use constructor(router: Router,
+   *  routingConfigService: RoutingConfigService,
```

Comment:
Add here `config` of type `CheckoutConfig`. We can't remove it before 2.x.x version and comment suggests using constructor which don't exists now. In 2.x.x we can get rid of this `config`. For now we have to keep it.